bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/rest/model/DnsStatistics.java

```diff
@@ -0,0 +1,540 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pihole.internal.rest.model;
+
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * @author Martin Grzeslowski - Initial contribution
+ */
+@NonNullByDefault
+public class DnsStatistics {
```

Comment:
I think you could take advantage of using `records` if this object is immutable.

Reply:
As far as I remember it does not work with provided version of Gson